### PR TITLE
Document new governance model

### PR DESCRIPTION
**Pull request by paulromano**
This PR modifies our documentation to include a contribution guide that is primarily intended to document a governance model. To summarize the governance model:

- Pull requests must be reviewed by a *committer* (subset of contributors who have write access)
- If multiple committers disagree on a PR, they must work together to reach a consensus on how to proceed
- If they can't reach consensus, a technical committee (TC) then has to make a decision, and they also use a consensus seeking process
- If the TC can't reach consensus, the project lead (me) has the final decision

The only change from what I originally discussed this week at our developer's meeting is that rather than a voting process being used when the TC can't reach consensus, the project lead has the final decision (since everyone seemed happy to have me as the benevolent dictator).

The TC would initially consist of me, @smharper, @nelsonag, and @bforget.

### CODE REVIEW

File: docs/source/devguide/contributing.rst [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+The default for each contribution is that it is accepted once no committer has
+an objection. During review committers may also request that a specific
+contributor who is most versed in a particular area review the PR before it can
+be merged. Once all issues brought by committers are addressed it can be merged
+by any committer.
+
+In the case of an objection being raised in a pull request by another committer,
+all involved committers should seek to arrive at a consensus by way of
+addressing concerns being expressed through discussion, compromise on the
+proposed change, or withdrawal of the proposed change.
```
Comment (smharper) [inline docs/source/devguide/contributing.rst]: I think this paragraph and the one above can be shortened and merged together with something like:

> A pull request can be merged by any committer, but only if no objections are raised by any other committer. In the case of an objection being raised, all involved committers should seek consensus through discussion and compromise.

That would remove the first sentence of the first paragraph which I think can be easily misunderstood (someone might think a merge is automatic at 36 hours if no review is made).  I think the bit about requesting a specific contributor (second sentence, first paragraph) will probably be understood without needing to be explicitly written down.

File: docs/source/devguide/contributing.rst [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+--------------------
+
+All contributors who make a non-trivial contribution will be added as a
+committer in a timely manner. Committers are expected to follow this policy.
+
+TC Process
+----------
+
+The TC uses a "consensus seeking" process for issues that are escalated to the
+TC. The group tries to find a resolution that has no objections among TC
```
Comment (smharper) [inline docs/source/devguide/contributing.rst]: The first sentence feels a little redundant with the word `TC`.  Consider rewording to something like

> Any issues brought to the TC will be addressed among the committee with a consensus-seeking process.

File: docs/source/devguide/contributing.rst [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+proceed, the decision is escalated to the TC. TC members should regularly
+discuss pending contributions in order to find a resolution. It is expected that
+only a small minority of issues be brought to the TC for resolution and that
+discussion and compromise among committers be the default resolution mechanism.
+
+Becoming a Committer
+--------------------
+
+All contributors who make a non-trivial contribution will be added as a
+committer in a timely manner. Committers are expected to follow this policy.
```
Comment (GiudGiud) [inline docs/source/devguide/contributing.rst]: Being picky, but should I read "Committers are expected to follow this policy." as all committers are in charge of promoting non-trivial contributors ?

Also there's no resign policy for committers ? It's not a big deal, just making sure it's not an omission

File: docs/source/devguide/contributing.rst [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  decision when the TC is unable to reach consensus.
+
+Contribution Process
+--------------------
+
+Any change to the OpenMC repository must be made through a pull request (PR).
+This applies to all changes to documentation, code, binary files, etc. Even long
+term committers and TC members must use pull requests.
+
+No pull request may be merged without being reviewed.
```
Comment (GiudGiud) [inline docs/source/devguide/contributing.rst]: I don't see a rule saying "no PR merged by its creator / review by the PR creator doesn't count as the required one". I'm not sure whether this was a rule before or not

File: docs/source/devguide/contributing.rst [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  to make decisions on behalf of the project team in order to resolve disputes.
+- The *Project Lead* is a single individual who has the authority to make a final
+  decision when the TC is unable to reach consensus.
+
+Contribution Process
+--------------------
+
+Any change to the OpenMC repository must be made through a pull request (PR).
+This applies to all changes to documentation, code, binary files, etc. Even long
+term committers and TC members must use pull requests.
```
Comment (GiudGiud) [inline docs/source/devguide/contributing.rst]: In the dev workflow guide : "Trivial changes to the code may be committed directly to the develop branch by a trusted developer." seems to conflict

Comment (paulromano) [reply docs/source/devguide/contributing.rst]: Ah, good point! I'll change that.

File: docs/source/devguide/contributing.rst (hunk 3 above)
Comment (paulromano) [reply docs/source/devguide/contributing.rst]: "Committers are expected to follow this policy" really is intended to say that committers do not obtain special rights with respect to the ability to change code -- they still have to submit PRs that undergo review.

No resign policy by design -- it's possible that someone might want to review a PR in an area of the code relevant to them long after they have been active.

File: docs/source/devguide/contributing.rst (hunk 4 above)
Comment (paulromano) [reply docs/source/devguide/contributing.rst]: I've added the word "independently" here.

File: docs/source/devguide/contributing.rst [hunk 6]
```diff
@@ -0,0 +1,116 @@
+.. _devguide_contributing:
+
+======================
+Contributing to OpenMC
+======================
+
+Thank you for considering contributing to OpenMC! We look forward to welcoming
+new members to the community and will do our best to help you get up to speed.
```
[the remark this answers is not in the document]
Comment (paulromano) [reply docs/source/devguide/contributing.rst]: I've added a new section at the end talking about "Next Steps". It mentions the First-Timers-Only issues as well as our developer's mailing list and Slack community.

**Pull request merged**